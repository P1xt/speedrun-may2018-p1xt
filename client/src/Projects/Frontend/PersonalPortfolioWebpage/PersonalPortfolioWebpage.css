.PersonalPortfolioWebpage {
  text-align: left;
  margin-top: 40px; }
  .PersonalPortfolioWebpage > h1 {
    text-transform: none; }
