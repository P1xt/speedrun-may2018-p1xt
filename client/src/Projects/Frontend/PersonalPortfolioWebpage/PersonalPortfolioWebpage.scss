.PersonalPortfolioWebpage {
  text-align: left;
  background: #ffffff;
  padding: 40px 60px;
  &>h1 {
    text-transform: none
  }

  section {
    margin-top: 40px;
  }
  &_about {
    display: flex;
    flex-direction: row;
    aside {
      margin-right: 50px;
      img {
        width: 200px;
      }
    }
    ul {
      display: flex;
      flex-direction: row;
      margin-left: -40px;
      li {
        list-style-type: none;
        margin-right: 30px;
      }
    }
    @media (max-width: 700px) {
      flex-direction: column;
      aside {
        display: none;
      }
    }
  }
  &_portfolio {
    border-radius: 10px;
      background: rgb(204, 204, 204);
      padding: 30px;
      .projects {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;
      justify-content: left;

      .card {
        margin-bottom: 30px;
      }
    }
  }
  
}
