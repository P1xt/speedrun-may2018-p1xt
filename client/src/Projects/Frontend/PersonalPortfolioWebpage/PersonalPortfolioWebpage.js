import React from "react";
import { Container, Row, Col } from 'reactstrap';

import "./PersonalPortfolioWebpage.css";

const PersonalPortfolioWebpage = () => {
  return (
    <div className="PersonalPortfolioWebpage">
      <Container>
        <Row>
          <Col>
            <h1>Personal Portfolio Webpage</h1>
          </Col>
        </Row>
      </Container>
    </div>
  );
};

export default PersonalPortfolioWebpage;
