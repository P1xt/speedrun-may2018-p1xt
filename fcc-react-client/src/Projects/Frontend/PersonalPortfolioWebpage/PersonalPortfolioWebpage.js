import React, { Component } from "react";
import {
  Container,
  Row,
  Col,
  Navbar,
  NavbarBrand,
  NavbarToggler,
  Collapse,
  NavItem,
  NavLink,
  Nav,
  Card,
  CardHeader,
  CardTitle,
  CardSubtitle,
  CardBody,
  CardImg,
  CardText
} from "reactstrap";

import "./PersonalPortfolioWebpage.css";

const projects = [
  {
    key: 1,
    name: "Tribute Page",
    image: "/images/tribute_page.png"
  },
  {
    key: 2,
    name: "Random Quote",
    image: "/images/random_quote_machine.png"
  },
  {
    key: 3,
    name: "Local Weather",
    image: "/images/local_weather.png"
  },
  {
    key: 4,
    name: "Wikipedia Viewer",
    image: "/images/wikipedia_viewer.png"
  },
  {
    key: 5,
    name: "Twitch TV",
    image: "/images/twitch_tv.png"
  },
  {
    key: 6,
    name: "Calculator",
    image: "/images/javascript_calculator.png"
  },
  {
    key: 7,
    name: "Pomodoro Clock",
    image: "/images/pomodoro_clock.png"
  },
  {
    key: 8,
    name: "Tic Tac Toe",
    image: "/images/tic_tac_toe.png"
  }
];
class PersonalPortfolioWebpage extends Component {
  constructor() {
    super();

    this.state = {
      isOpen: false
    };
  }

  toggle = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  };
  render() {
    return (
      <div>
        <Navbar color="light" light expand="md">
          <NavbarBrand href="/">Portfolio</NavbarBrand>
          <NavbarToggler onClick={this.toggle} />
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav className="ml-auto" navbar>
              <NavItem>
                <NavLink href="#about">About</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="#portfolio">Portfolio</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="#contact">Contact</NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
        <Container>
          <div class="PersonalPortfolioWebpage">
            <section class="PersonalPortfolioWebpage_about" id="about">
              <div>
                <Row>
                  <Col xs="12">
                    <h1>P1xt</h1>
                    <p>
                      Freelance Software Engineer specializing in full stack
                      JavaScript Web development.
                    </p>
                  </Col>
                  <Col xs="12">
                    <h2>Current Focus</h2>
                    <hr />
                    <h3>Professionally</h3>
                    <ul>
                      <li>Angular</li>
                      <li>Node.js</li>
                      <li>Ionic</li>
                    </ul>

                    <h3>Speedrun</h3>
                    <ul>
                      <li>React.js</li>
                      <li>Node.js</li>
                      <li>Phaser.js</li>
                    </ul>
                  </Col>
                </Row>
              </div>
            </section>
            <section class="PersonalPortfolioWebpage_portfolio" id="portfolio">
              <h1>Portfolio</h1>
              <p>Examples of my recent work:</p>
              <div class="projects">
                <Row>
                  {projects.map(project => (
                    <Col key={project.key} xs="12" md="6">
                      <Card>
                        <CardImg
                          top
                          width="100%"
                          alt="project"
                          src={project.image}
                        />
                        <CardHeader>
                          <CardTitle>{project.name}</CardTitle>
                          <CardSubtitle>May 2018 Speedrun</CardSubtitle>
                        </CardHeader>
                        <CardBody>
                          <CardText>
                            Implementation of the Free Code Camp {project.name}{" "}
                            project in React.js for May 2018 Speedrun.
                          </CardText>
                        </CardBody>
                      </Card>
                    </Col>
                  ))}
                </Row>
              </div>
            </section>
            <section class="PersonalPortfolioWebpage_contact" id="contact">
              <h1>Contact</h1>
              <p>
                You can get ahold of me on{" "}
                <a href="https://twitter.com/P1xt">Twitter</a> and sometimes I
                write some stuff on{" "}
                <a href="https://medium.com/@P1xt">Medium</a>.
              </p>
              <p>
                Though, truth be told, this is a throwaway project for a
                speedrun I'm doing so caveat emptor - if you're using this to
                contact me, you're probably going to be disappointed.
              </p>
            </section>
          </div>
        </Container>
      </div>
    );
  }
}

export default PersonalPortfolioWebpage;
