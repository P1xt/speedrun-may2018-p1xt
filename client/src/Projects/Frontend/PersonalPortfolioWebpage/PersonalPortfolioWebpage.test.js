import React from "react";
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";

import PersonalPortfolioWebpage from "./PersonalPortfolioWebpage";

describe("<PersonalPortfolioWebpage />", () => {
  it("renders without crashing", () => {
    shallow(<PersonalPortfolioWebpage />);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<PersonalPortfolioWebpage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
