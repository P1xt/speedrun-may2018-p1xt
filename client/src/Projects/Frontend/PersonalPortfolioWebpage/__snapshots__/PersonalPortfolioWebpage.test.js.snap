// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalPortfolioWebpage /> renders correctly 1`] = `
<div
  className="PersonalPortfolioWebpage"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h1>
          Personal Portfolio Webpage
        </h1>
      </div>
    </div>
  </div>
</div>
`;
